htrace-core4/src/test/java/org/apache/htrace/core/TestNewScopeWithParentID.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.htrace.core;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestNewScopeWithParentID {
+
+  @Test
+  public void testNewScopeWithParentID() throws Exception {
+
+    Tracer tracer = new Tracer.Builder().
+              name("testNewScopeWithParentID").
+              tracerPool(new TracerPool("testNewScopeWithParentID")).
+              conf(HTraceConfiguration.fromKeyValuePairs(
+                      "sampler.classes", "AlwaysSampler")).build();
+    TraceScope activeScope = tracer.newScope("CurrentActiveScope");
+    SpanId parentID = new SpanId(100L, 200L);
+    TraceScope childScope = tracer.
+              newScope("ChildScope", parentID);
+    Assert.assertNotNull(childScope);
+    Assert.assertEquals(childScope.getSpan().getParents().length, 2);
+    Assert.assertEquals(childScope.getSpan().getParents()[1].getHigh(), 100L);
```

Comment:
Is the order guaranteed? I'm not sure that it is, rather than being an implementation detail in this case.